### PR TITLE
[AgentWorker.PageLogger] - Ignore endgroup tag when counting lines

### PR DESCRIPTION
**Pull request by DmitriiBobreshev**
**Description**
- Added tests for the new logic
- Added logic to ignore endgroup tag when counting lines for the error/warning highlight as it's not shown in the UI it shouldn't be counted

**Testing**
Tested via pipeline and added unit tests

**Related Issue**: WI 2125343

### CODE REVIEW

**Review by vmapetr — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

File: src/Test/L0/PagingLoggerL0.cs [hunk 1]
```diff
@@ -11,6 +11,11 @@ namespace Microsoft.VisualStudio.Services.Agent.Tests.Listener
     public sealed class PagingLoggerL0
     {
         private const string LogData = "messagemessagemessagemessagemessagemessagemessagemessageXPLATmessagemessagemessagemessagemessagemessagemessagemessage";
+        private const string LogDataWithGroup = @"messagemessagemessagemes
```
Comment (DergachevE) [inline src/Test/L0/PagingLoggerL0.cs:14]: Should we add edge cases (e.g., no start group tag but end tag)?
